src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1021,6 +1035,33 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
                                 summary='Problem loading the helm existing user supplied values')
         raise CLIInternalError("Problem loading the helm existing user supplied values: " + str(e))
 
+    # Change registry_path to private registry if PCR is set in helm values
+    existing_user_values_flat = utils.flatten(existing_user_values)
+    helm_container_repository = existing_user_values_flat.get('systemDefaultValues.image.repository')
+    registry_username = None
+    registry_password = None
+    if helm_container_repository.split('.')[0] != "mcr":
+        registry_path = "{}/azurearck8s/azure-arc-k8sagents:{}".format(helm_container_repository, arc_agent_version)
+        registry_username = existing_user_values_flat.get('systemDefaultValues.image.username')
+        registry_password = existing_user_values_flat.get('systemDefaultValues.image.password')
+        # Send private registry information to telemetry
+        telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.PrivateContainerRegistry': helm_container_repository})
+    else:
+        # Retrieving Helm chart OCI Artifact location
+        registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+
+    reg_path_array = registry_path.split(':')
+    agent_version = reg_path_array[1]
+
+    if arc_agent_version is not None:
+        agent_version = arc_agent_version
+        registry_path = reg_path_array[0] + ":" + agent_version
+
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.AgentVersion': agent_version})
```

Comment:
If this agent version based on PCR, we should tag it in telemetry somehow. Since, these agent versions will be maintained privately by the customers, and not us.

Reply:
i think the telemetry for PCR registry is being collected above

Reply:
We are having a telemetry for PCR and also we are sending agent version so I dont think there is separate need of tagging this